includes/class-wc-payments-utils.php

```diff
@@ -1251,74 +1251,31 @@ public static function enqueue_style( $handle, $path = '', $deps = [], $version
 	}
 
 	/**
-	 * Returns language data: english name and native name
+	 * Converts a WP locale to a wpcom-compatible language code.
 	 *
-	 * @param string $language Language code.
+	 * @see Automattic\Jetpack\Jetpack_Mu_Wpcom\Common::get_iso_639_locale() for similar logic.
+	 * @see https://translate.wordpress.com/projects/wpcom/ for the current state of wpcom translations.
 	 *
-	 * @return array
-	 */
-	public static function get_language_data( $language ) {
-		require_once ABSPATH . 'wp-admin/includes/translation-install.php';
-
-		$translations = wp_get_available_translations();
-
-		if ( isset( $translations[ $language ] ) ) {
-			return [
-				'code'         => self::convert_to_server_locale( $language ),
-				'english_name' => $translations[ $language ]['english_name'] ?? $language,
-				'native_name'  => $translations[ $language ]['native_name'] ?? $language,
-			];
-		}
-
-		return [
-			'code'         => 'en_US',
-			'english_name' => 'English (United States)',
-			'native_name'  => 'English (United States)',
-		];
-	}
-
-	/**
-	 * Converts a locale to the server supported languages.
-	 *
-	 * @param string $locale The locale to convert.
-	 *
-	 * @return string Closest locale supported ('en' if NONE)
+	 * @param string $wp_locale a WordPress locale code to be converted e.g. "en_US", "pt_BR".
+	 * @return string language code compatible with wpcom e.g. "en", "pt-br".
 	 */
-	public static function convert_to_server_locale( string $locale ): string {
-		$supported = [
-			'ar',     // Arabic.
-			'de',     // German (Germany).
-			'es',     // Spanish (Spain).
-			'fr',     // French (France).
-			'he',     // Hebrew (Israel).
-			'id',     // Indonesian (Indonesia).
-			'it',     // Italian (Italy).
-			'ja',     // Japanese.
-			'ko',     // Korean.
-			'nl',     // Dutch (Netherlands).
-			'pt-br',  // Portuguese (Brazil).
-			'ru',     // Russian (Russia).
-			'sv',     // Swedish (Sweden).
-			'tr',     // Turkish (Turkey).
-			'zh-cn',  // Simplified, Singapore).
-			'zh-tw',  // Chinese Traditional (Taiwan).
-		];
-
-		// Replace '-' with '_' (used in WordPress).
-		$locale = str_replace( '_', '-', $locale );
-
-		if ( in_array( $locale, $supported, true ) ) {
-			return $locale;
-		}
-
-		// Remove the country code and try with that.
-		$base_locale = substr( $locale, 0, 2 );
-		if ( in_array( $base_locale, $supported, true ) ) {
-			return $base_locale;
+	public static function convert_to_server_locale( string $wp_locale ): string {
+		$wp_locale_lowercase                    = strtolower( $wp_locale );
+		$region_specific_wpcom_language_codes   = [ 'pt_br', 'pt-br', 'zh_tw', 'zh-tw', 'zh_cn', 'zh-cn' ];
```

Comment:
Reducing the manual mapping of locales only to those that are region-specific.

This approach is borrowed from [Jetpack's `get_iso_639_locale()`￼](https://github.com/Automattic/jetpack/blob/0ed6e32cb2a5c4c0f7a34d76a8606fedf7ba8321/projects/packages/jetpack-mu-wpcom/src/common/index.php#L19C1-L39C2)

Reply:
> This approach is borrowed from [Jetpack's get_iso_639_locale()￼](https://github.com/Automattic/jetpack/blob/0ed6e32cb2a5c4c0f7a34d76a8606fedf7ba8321/projects/packages/jetpack-mu-wpcom/src/common/index.php#L19C1-L39C2)

Recommend adding a comment "maintenance hint" about this, i.e. guide future team when WPCOM adds a new language (or removes one).

Reply:
Maybe in the docblock.